File: Tp1/Dona Monica/Dona.java
import java.util.Scanner;
public class Dona{
	public static int maisVelho(int n1, int n2, int n3){
		int n4 = n1 - (n2 + n3);
		if(n3 > n2 && n3 > n4){
			return n3;
		} else if(n2 > n3 && n2 > n4){
			return n2;
		} else {
			return n4;
		}
	}

	public static boolean isFim(int n){
		return (n == 0);
	}

	public static void main(String args[]){
		Scanner entrada = new Scanner(System.in);

		int n1 = entrada.nextInt();
		int n2 = 0;
		int n3 = 0;
		do{
			n2 = entrada.nextInt();
			n3 = entrada.nextInt();
			if(n3 + n2 < n1 ){
				System.out.println(maisVelho(n1, n2, n3));
			} else {
				break;
			}
			n1 = entrada.nextInt();
		}while(!isFim(n1));
	}
}

File: Tp1/Dona Monica/Dona.c
#include <stdio.h>
#include <stdbool.h>

int maisVelho(int n1, int n2, int n3){
	int n4 = n1 - (n2 + n3);
	if(n3 > n2 && n3 > n4){
		return n3;
	} else if(n2 > n3 && n2 > n4){
		return n2;
	} else {
		return n4;
	}
}

bool isFim(int n){
	return (n==0);
}

int main(void){
	int n1,n2,n3 = 0;
	
	scanf("%d", &n1);
	do{
		scanf("%d", &n2);
		scanf("%d", &n3);
		if(n3 + n2 < n1){
			printf("%d", maisVelho(n1, n2, n3));
		} else {
			break;
		}
		scanf("%d", &n1);
	}while(!isFim(n1));

	return 0;
}
